### ISSUE TITLE
Game crashes after several inputs

### ISSUE THREAD

**Issue by ewaynem** [LordxOsmosisx86/DMG-KGDU-10-Plus-FRAM-Upgraded-PCB#3]
BOM:

- 1x   DMG-KGDU-10+, from JLCPCB
- 1x   FM18W08SG
- 1x   MX23C1603-12A (DMG-AAXE-0), from original Silver cartridge
- 1x   MBC3-B, from original Silver cartridge
- 1x   M030 134A, from original Silver cartridge
- 1x   R26-32.768-12.5-10PPM, from original Silver cartridge
- 5x   187-CL10B104KO8NNNC, from Mouser
- 2x   187-CL10C150JB8NNNC, from Mouser
- 1x   603-RC0603FR-07330KL, from Mouser
- 1x   603-RC0603FR-0710KL, from Mouser
- 1x   603-RC0603FR-071KL, from Mouser
- 1x   74LVC1G32GV,125, from Mouser
- 1x   712-BAT-6219-TR, from Digikey
- 1x   CR2025 (Energizer)

I have tested this on two individual boards, one running an original Pokemon Silver rom, and one running an original Trade & Battle: Card Hero rom.  I have not tested the RTC on the Card Hero, but the RTC on the SIlver functions properly and has been running for several days without failure.  The FRAM chip on the Silver functions and holds a save.

Each part was soldered on using hot air and flux.  They were placed in one at a time to prevent mix up.  I have tested the resistors to insure they are not in the incorrect positions.  I do not have a way to test the capacitors, though I am certain they are placed correctly.  Touchup was done with a soldering iron and more flux.  There are no bridges.

Sometimes the games run for a minute before crashing, sometimes they crash before getting past the title screen.  They will both either freeze entirely, crash to a white screen, or reset to the title menu with corrupted sprites and colors.  Most often they crash after A or B inputs or D-pad inputs.  This occurs in both an original Gameboy Color and an original Gameboy Advance SP.  I haven't yet tested whether this occurs in a Gameboy Advance or a Gamecube Gameboy Player.

![IMG_7573](https://user-images.githubusercontent.com/128922689/227732759-2e81c643-9995-4025-91fe-366cc7613860.png)
![IMG_7574](https://user-images.githubusercontent.com/128922689/227732758-f5bcaef2-18a3-45d5-b98d-008471c212a8.png)
![IMG_7575](https://user-images.githubusercontent.com/128922689/227732755-fd5fdd9e-a53b-4e88-ba46-06186ed688f9.png)

**Comment by Mhouranix1125** [comment]
One thing that usually runs consistent when people run into issues attempting to move things from an OEM board to one of my boards is the initial use of solder paste and hot air.

Upon inspecting your board, it also looks like you were pretty rough with the iron and scraped around while soldering in the chip (if you look at some of the pads you can see you've dug up and tinned the traces in several areas). 

When I move things to this board, I have found that you should only use solder paste and hot air on resistors and capacitors. I always put those in first. 

Next I only use a soldering iron and Flux to put the chips in. The issues you're talking about pertaining to the white screens amd crashing usually happen when the FRAM is not soldered in properly.

**Comment by Mhouranix1125** [comment]
Also. What or gate are you using right there? It has a Y32 symbol on it from what I can see. Can you send me the link you used to purchase that?

Send me exactly what you ordered from mouser and everything else. I need the exact model of it all. I am suspicious of the or gate you're using.

**Comment by ewaynem** [comment]
Here are the links to the parts I ordered:

https://www.mouser.com/ProductDetail/771-LVC1G32GV125
https://www.mouser.com/ProductDetail/187-CL10B104KO8NNNC
https://www.mouser.com/ProductDetail/187-CL10C150JB8NNNC
https://www.mouser.com/ProductDetail/603-RC0603FR-07330KL
https://www.mouser.com/ProductDetail/603-RC0603FR-0710KL
https://www.mouser.com/ProductDetail/603-RC0603FR-071KL

**Comment by ewaynem** [comment]
Also, I tested the Pokemon Silver on the Gamecube Gameboy Player for about ten minutes or so, and it didn't crash.  This is the longest the board went without crashing.

**Comment by Mhouranix1125** [comment]
Those parts look fine. 

The issue is likely in the solder work. I don't know your level of expertise but I see a lot of solder mask scraped off near pads on both boards. I see some traces that are exposed and tinned. On the card game, why is the board all white on the left hand side? 

I would reattempt this on fresh boards and I wouldn't use any solder paste. Your iron may be a little too hot as well and you may be pressing a bit hard.

**Comment by Mhouranix1125** [comment]
Can you send me the specifications of the board you ordered from JLC? Are they the correct thickness?

**Comment by ewaynem** [comment]
![Screen Shot 2023-03-25 at 6 59 58 PM](https://user-images.githubusercontent.com/128922689/227748004-d2734eb0-b777-4585-ace7-0e7aefed17ee.png)

The card game board is white because of the glare from the light.

**Comment by Mhouranix1125** [comment]
Got it. Reattempt this on fresh boards. Don't use solder paste.

If it doesn't work you can send one or both games to me and I will put them together for you at no cost other than shipping. I will require 2 fresh boards from you on your end. This will help rule out issues with your board.

If I find that your boards are faulty I will use two of my own and ensure you have working games. 

Let me know if this interests you. I highly encourage you try it out again on your own though.

**Comment by Mhouranix1125** [comment]
@ewaynem checking in. Have you attempted the transfer to a new board?

**Comment by ewaynem** [comment]
I have attempted a new board with minimal to no success.  Now getting a corrupted Nintendo logo at the Gameboy screen, which I'm certain is a result of my, shall we say, limited soldering skills.  I even grabbed another mapper chip to ensure that it wasn't a hardware issue.  I've inspected the pins of each chip and none of them are loose.

![IMG_7580](https://user-images.githubusercontent.com/128922689/227827659-bc89c6e3-b864-4da3-8e9d-751cebec30e5.png)

**Comment by Mhouranix1125** [comment]
> I have attempted a new board with minimal to no success. Now getting a corrupted Nintendo logo at the Gameboy screen, which I'm certain is a result of my, shall we say, limited soldering skills. I even grabbed another mapper chip to ensure that it wasn't a hardware issue. I've inspected the pins of each chip and none of them are loose.
> 
> ![IMG_7580](https://user-images.githubusercontent.com/128922689/227827659-bc89c6e3-b864-4da3-8e9d-751cebec30e5.png)

Yes. The corrupted nintendo logo can happen because chips are incorrectly soldered. Your MBC3 looks like it needs more solder. It doesn't look like it's making a reliable connection. Where did you get the FRAM chips from?

You shouldn't be attempting this if you're pretty new to soldering, though.

It might be a better idea to practice on some cheap throw away games. I would buy two copies of a game like bugs life and practice transferring rom chips over. 

Or you can buy Japanese versions of pokemon gold and silver and swap mask roms until you get it right.

**Comment by ewaynem** [comment]
That's the funny thing, I have swapped roms and SRAM and FRAM chips before with great success.  I got the FRAM chips from Aliexpress and ebay.

**Comment by Mhouranix1125** [comment]
> That's the funny thing, I have swapped roms and SRAM and FRAM chips before with great success. I got the FRAM chips from Aliexpress and ebay.

If you ever feel like letting me take a swing at helping you with this, let me know.
